drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -338,20 +345,19 @@ end
 --- In this case the function returns the lowest endpoint value that isn't 0
 --- and supports the OnOff or Switch cluster. This is done to bypass the
 --- BRIDGED_NODE_DEVICE_TYPE on bridged devices.
-local function find_default_endpoint(device, component)
+local function find_default_endpoint(device)
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
   local button_eps = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.Feature.MOMENTARY_SWITCH})
-  local all_eps = {}
+  table.sort(switch_eps)
+  table.sort(button_eps)
 
+  -- Use the first switch endpoint as the main endpoint if one is present.
```

Comment:
One thing to mention about this strategy is that if a switch endpoint is going to be used first, that means anytime we encounter a device that is a switch+button combination that we don't currently support in our profiles, then we would have to create a new profile to cover that switch+button combination. For example, right now we have light level switch + N buttons supported in our profiles, but let's say another device comes out that is just on/off switch + buttons. Then we would have to create new permutations of the existing button profiles.

We can continue to make profiles, but again this is something where we will have to know what products are on the market and create new profiles as needed to cover the capabilities of those new switch+button combos. This is something we can do with profiles, but dynamic devices would greatly reduce this burden when that is available.

Reply:
@nickolas-deboom we are making the switch endpoint the main component for UX reasons, right?

If we didn't do that, we could decouple the switch endpoints from the button endpoints in the profiles entirely. However, I understand this could provide a bad user experience for customers if they have all their buttons in one device card and then an on/off switch in a separate one.

Reply:
Unfortunately you are right, that new profiles would be needed for new products if they use different device types than what is currently supported.

Reply:
And yes that's correct - after some conversations with Tom, it would be best from a UX perspective to keep the buttons with the light device in the same device card. This also matches the way that zigbee implements a similar Inovelli device.